src/tools/pub/cmd/pub-outdated.md

```diff
@@ -0,0 +1,248 @@
+  [TODO: Improve the formatting of the following diff]
+{% endcomment %}
+
+```diff
+-  build_web_compilers: ^1.0.0
++  build_web_compilers: ^2.10.0
+```
+
+{{site.alert.tip}}
+  To see what changed in the new version of a package
+  that's published on [pub.dev,]({{site.pub}})
+  look at the changelog in the package page.
+  For an example, see the
+  [build_web_compilers changelog.][]
+{{site.alert.end}}
+
+After editing `pubspec.yaml`, run `pub upgrade` to
+update the `pubspec.lock` file.
+You can then run `pub outdated` to confirm that
+you've made all necessary changes.
+
+```terminal
+$ pub upgrade
+...
+$ pub outdated
+Package     Current  Upgradable  Resolvable  Latest  
+dependencies: all up-to-date
+
+dev_dependencies: all up-to-date
+
+transitive dependencies
+googleapis  0.53.0   0.53.0      0.53.0      0.54.0  
+$
+```
+
+{{site.alert.important}}
+  Test your code to verify that it still works as expected
+  after updating the packages.
+{{site.alert.end}}
+
+If any transitive dependencies are out-of-date,
+ask the package maintainers to update their code.
+To find the cause of an out-of-date dependency,
+you can run [`pub deps`][] and
+search the output for the name of the out-of-date package.
+Then make sure an issue exists for the out-of-date dependency.
+You can often find the issue tracker for a package
+by looking at the package page on pub.dev.
+
+
+## Output columns
+
+As a reminder, here's what the output of `pub outdated` looks like:
+
+
+```terminal
+$ pub outdated
+Package              Current   Upgradable  Resolvable  Latest    
+dependencies: all up-to-date
+
+dev_dependencies    
+build_web_compilers  1.2.3     1.2.3       2.10.0      2.10.0    
+
+transitive dependencies
+_fe_analyzer_shared  -         -           2.0.0       2.0.0     
+analyzer             0.35.4    0.35.4      0.39.5      0.39.5    
+...
+
+9  dependencies are constrained to versions that are older than a resolvable version.
+To update these dependencies, edit pubspec.yaml.
+```
+
+As a rule, update the **dependencies** and **dev_dependencies** sections
+of your `pubspec.yaml` file
+to depend on the **resolvable** versions of out-of-date packages.
+
+Here are the meanings of the version numbers in the output:
+
+Current
+: The version used in your package, as recorded in `pubspec.lock`.
+  If the package isn't in `pubspec.lock`,
+  the value is `-`.
+
+Upgradable
+: The latest version allowed by your `pubspec.yaml` file.
+  This is the version that `pub upgrade` resolves to.
+  The value is `-` if the value in the **Current** column is `-`.
+
+  <span style="color:red">
+    [PENDING: Is the description of `-` correct?
+    Is there a more helpful way to state it?]
+  </span>
+
+Resolvable
+: The latest version that can be resolved.
+  This version corresponds to what you get if
+  you set all version constraints in `pubspec.yaml` to `any`
+  and then run `pub upgrade`.
+  A value of `-` means that the package will no longer be needed.
```

Comment:
I'm not sure how we want to phrase this. This is not a bad explanation, but we don't explain why we're showing this result.

**reasoning**: We compute the `latest resolvable version` because this is the latest version **you can get** in combination with the other dependencies you have.

So while you may not be able to get the latest versions of all packages, you can take the version number from `resolvable` column and insert that in `pubspec.yaml` with a `^` in-front.. and if you do that for all versions in the resolvable column then version resolution will work. So in cases where you can't get the latest version, _resolvable_ helps you discover what the latest version _you can get_ is.